Error handling for dbt retry (#3985)
fishtown-analytics/docs.getdbt.com: website/docs/reference/commands/retry.md

`dbt retry` reuses the [selectors](/reference/node-selection/yaml-selectors) from the previously executed command.

> If you run `dbt retry` after successfully executing `dbt run`, you'll get the following results.

Author: ```suggestion
Executing `dbt retry` after a successful `dbt run` will yield the following results:
```

3 of 5 START sql view model main.stg_payments .................................. [RUN]
1 of 5 OK created sql view model main.stg_customers ............................ [OK in 0.06s]
2 of 5 OK created sql view model main.stg_orders ............................... [OK in 0.06s]
3 of 5 OK created sql view model main.stg_payments ............................. [OK in 0.07s]
4 of 5 START sql table model main.customers .................................... [RUN]
5 of 5 START sql table model main.orders ....................................... [RUN]
4 of 5 ERROR creating sql table model main.customers ........................... [ERROR in 0.03s]
5 of 5 OK created sql table model main.orders .................................. [OK in 0.04s]
 
Finished running 3 view models, 2 table models in 0 hours 0 minutes and 0.15 seconds (0.15s).
  
Completed with 1 error and 0 warnings:
  
Runtime Error in model customers (models/customers.sql)
 Parser Error: syntax error at or near "selct"

Done. PASS=4 WARN=0 ERROR=1 SKIP=0 TOTAL=5
```

Executing `dbt retry` without fixing the error(s) will result in another failed run:

> ```

Author: ```suggestion

```
```

`dbt retry` reuses the [selectors](/reference/node-selection/yaml-selectors) from the previously executed command.

> Executing `dbt retry` after a successful `dbt run` will yield the following results:

Reviewer: ```suggestion
Example results of executing `dbt retry` after a successful `dbt run`:
```

> ```

Reviewer: ```suggestion
```shell
```

Running with dbt=1.6.1
Registered adapter: duckdb=1.6.0
Found 5 models, 3 seeds, 20 tests, 0 sources, 0 exposures, 0 metrics, 348 macros, 0 groups, 0 semantic models
 
Nothing to do. Try checking your model configs and model specification args
```

> In a scenario where `dbt run` has failed due to a syntax error in a model:

Reviewer: ```suggestion
Example of when `dbt run` encounters a syntax error in a model: 
```

> ```

Reviewer: ```suggestion
```shell
```

2 of 5 START sql view model main.stg_orders .................................... [RUN]
3 of 5 START sql view model main.stg_payments .................................. [RUN]
1 of 5 OK created sql view model main.stg_customers ............................ [OK in 0.06s]
2 of 5 OK created sql view model main.stg_orders ............................... [OK in 0.06s]
3 of 5 OK created sql view model main.stg_payments ............................. [OK in 0.07s]
4 of 5 START sql table model main.customers .................................... [RUN]
5 of 5 START sql table model main.orders ....................................... [RUN]
4 of 5 ERROR creating sql table model main.customers ........................... [ERROR in 0.03s]
5 of 5 OK created sql table model main.orders .................................. [OK in 0.04s]
 
Finished running 3 view models, 2 table models in 0 hours 0 minutes and 0.15 seconds (0.15s).
  
Completed with 1 error and 0 warnings:
  
Runtime Error in model customers (models/customers.sql)
 Parser Error: syntax error at or near "selct"

Done. PASS=4 WARN=0 ERROR=1 SKIP=0 TOTAL=5
```

> Executing `dbt retry` without fixing the error(s) will result in another failed run:

Reviewer: ```suggestion
Example of a subsequent failed `dbt retry` run without fixing the error(s):
```

> ```

Reviewer: ```suggestion
```shell
```

Running with dbt=1.6.1
Registered adapter: duckdb=1.6.0
Found 5 models, 3 seeds, 20 tests, 0 sources, 0 exposures, 0 metrics, 348 macros, 0 groups, 0 semantic models

Concurrency: 24 threads (target='dev')

1 of 1 START sql table model main.customers .................................... [RUN]
1 of 1 ERROR creating sql table model main.customers ........................... [ERROR in 0.03s]

Done. PASS=4 WARN=0 ERROR=1 SKIP=0 TOTAL=5
```

> Executing `dbt retry` after a failed run, but correcting the issue, will result in a completed run:

Reviewer: ```suggestion
Example of a successful `dbt retry` run after fixing error(s):
```

> ```

Reviewer: ```suggestion
```shell
```
